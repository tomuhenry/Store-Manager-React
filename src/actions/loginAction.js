import actionTypes from './actionTypes';
import axios from 'axios';

export const fetchLoginSuccess = payload => ({
  type: actionTypes.LOGIN_SUCCESS,
  payload: payload
});

export const fetchLoginFailure = error => ({
  type: actionTypes.LOGIN_FAIL,
  error
});

const loginAction = userData => dispatch => {
  return axios
    .post(
      'https://tomuhenry-storemanagerapp.herokuapp.com/store-manager/api/v1/auth/login',
      userData
    )
    .then(response => {
      if (response.data.user_token) {
        dispatch(fetchLoginSuccess(response.data));
        window.localStorage.setItem('access_token', response.data.user_token);
      } else if (response.data.admin_token) {
        dispatch(fetchLoginSuccess(response.data));
        window.localStorage.setItem('access_token', response.data.admin_token);
      } else if (response.data.Alert) {
        document.getElementById('message').innerHTML = `${response.data.Alert}`;
        dispatch(fetchLoginFailure(response.data));
      }
    })
    .catch(error => {
      document.getElementById('message').innerHTML =
        'Login failed! check login info';
      dispatch(fetchLoginFailure(error.response));
    });
};

export default loginAction;
