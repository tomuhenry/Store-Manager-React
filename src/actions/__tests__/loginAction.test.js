import configureMockStore from 'redux-mock-store';
import React from 'react';
import thunk from 'redux-thunk';
import mockAxios from 'jest-mock-axios';
import "babel-polyfill";
import { shallow } from 'enzyme';
import loginAction from '../loginAction';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({});

describe('Social Actions', () => {
  afterEach = () => {
    mockAxios.reset();
  };

  it('dispatch actions for login', () => {
    const userData = { email: "email@here.com", passwprd:"my password" }
    mockAxios.post('MockURL', userData);
    store.dispatch(loginAction);
    expect(store.getActions()).toEqual([]);
  });

  it('dispatch actions for google login', () => {
    let wrapper = shallow(<loginAction />);
    expect(wrapper).toMatchSnapshot();
  });

});
