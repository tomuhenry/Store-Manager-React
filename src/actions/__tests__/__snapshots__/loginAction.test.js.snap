// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Actions dispatch actions for login 1`] = `<loginAction />`;
